src/Command/ModuleDeployAfterMagentoCommand.php

```diff
@@ -43,83 +112,152 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $startTime = microtime(true);
 
-        $projectRoot = getcwd();
-        $directoryName = basename($projectRoot);
+        $mainDomain = basename(getcwd());
+        $modulesFolder = $input->getArgument(self::OPTION_FOLDER);
+
+        $together = $input->getOption(self::OPTION_TOGETHER);
 
-        # Stage 1: copy files, deploy Sample Data, create database
-        if (!file_exists('./app') || !is_dir('./app')) {
-            $output->writeln('<error>Seems we\'re not inside the Magento project</error>');
-            exit;
+        if (file_exists('./composer.json')) {
+            $magentoComposerFile = json_decode(file_get_contents('./composer.json'));
+
+            if (!is_dir('./app') || strpos($magentoComposerFile->name, 'magento/') === false) {
+                $output->writeln('<error>Seems we\'re not inside the Magento project</error>');
+                exit(1);
+            }
+        } else {
+            $output->writeln('<error>Composer.json not found</error>');
+            exit(1);
         }
 
-        // @TODO: use composer require to install from the repository
-        // @TODO: add directory parameter with all modules
-        // @TODO: find all etc/module.xml, get vendor-module, create folders, copy
+        if (!is_dir($modulesFolder)) {
+            $output->writeln('<error>Modules directory does exist.</error>');
+            exit(1);
```

Comment:
@serhiira Must return exit code instead of actually exiting. Using the `exit` function terminates the process and makes it impossible to call this command from another command. This should be done only in case you really need to exit - like in the `hardware:test` command where child processes must be terminated.